validation/validate_dc2_run2.2i_object_table.ipynb

```diff
@@ -0,0 +1,1367 @@
+    "2. Distribution of ellipticity (see validate_drp to type this right)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def ellipticity(I_xx, I_xy, I_yy):\n",
+    "    \"\"\"Calculate ellipticity from second moments.\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
+    "    I_xx : float or numpy.array\n",
+    "    I_xy : float or numpy.array\n",
+    "    I_yy : float or numpy.array\n",
+    "\n",
+    "    Returns\n",
+    "    -------\n",
+    "    e, e1, e2 : (float, float, float) or (numpy.array, numpy.array, numpy.array)\n",
+    "        Complex ellipticity, real component, imaginary component\n",
+    "        \n",
+    "    Copied from https://github.com/lsst/validate_drp/python/lsst/validate/drp/util.py\n",
+    "    \"\"\"\n",
+    "    e = (I_xx - I_yy + 2j*I_xy) / (I_xx + I_yy + 2*SM.sqrt(I_xx*I_yy - I_xy*2))\n",
+    "    e1 = np.real(e)\n",
+    "    e2 = np.imag(e)\n",
+    "    return e, e1, e2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def plot_ellipticity(df, filt, good_idx, star_idx, galaxie_idx, ax=None, legend=True):\n",
+    "    if not ax:\n",
+    "        ax = fig.gca()\n",
+    "\n",
+    "    names = ['good', 'star', 'galaxy']\n",
+    "    colors = ['blue', 'orange', 'green']\n",
+    "    hist_kwargs = {'color': colors, 'log': True,\n",
+    "             'bins': np.logspace(-1, 1.5, 100),\n",
+    "             'range': (0, 5),\n",
+    "             'histtype': 'step'}\n",
+    "    for prefix, ls in (('e', '-'), ('e1', '--'), ('e2', ':')):\n",
+    "        field = f'{prefix}_{filt}'\n",
+    "        labels = [f'{prefix} {name}' for name in names]\n",
+    "        ax.hist([df.loc[good_idx, field], df.loc[star_idx, field], df.loc[galaxy_idx, field]],\n",
+    "                label=labels,\n",
+    "                linestyle=ls,\n",
+    "                **hist_kwargs)\n",
+    "\n",
+    "    ax.set_xlim(0, 20)\n",
+    "    ax.set_ylim(10)\n",
+    "    \n",
+    "    ax.set_xlabel(f'{filt}-band ellipticity')\n",
+    "    ax.set_ylabel('objects / bin')\n",
+    "    if legend:\n",
+    "        ax.legend()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for filt in filters:\n",
+    "    df[f'e_{filt}'], df[f'e1_{filt}'], df[f'e2_{filt}'] = \\\n",
+    "    ellipticity(df[f'Ixx_{filt}'], df[f'Ixy_{filt}'], df[f'Iyy_{filt}'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig, axes = plt.subplots(2, 3, figsize=(12, 6))\n",
+    "legend = True\n",
+    "for ax, filt in zip(axes.flat, filters):\n",
+    "    plot_ellipticity(df, filt, good_idx, star_idx, galaxy_idx, ax=ax, legend=legend)\n",
+    "    legend = False"
```

Comment:
After the for loop, add `plt.tight_layout()` so that the plot looks nicer (less overlapping text)